### PR TITLE
MM-40114/MM-40545 Clean up MobileSidebarHeader and have it use getIsMobileView

### PR DESCRIPTION
**Pull request by hmhealey**
There's a few things going on here because I found some other things to fix while switching some usage of `Utils.isMobile` to `getIsMobileView`:
1. I renamed the LegacySidebarHeader to MobileSidebarHeader since its only used in mobile view
2. I removed a bunch of unused logic from it that would never appear because the logic was for desktop view, but the component is only mounted in mobile view
3. I moved the keyboard handler for the ctrl/cmd+shift+a hotkey out of it so that it would work again in desktop view. I also added an E2E test for this (MM-40545)

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-40114
https://mattermost.atlassian.net/browse/MM-40545

#### Release Note
```release-note
Fixed Account Settings keyboard shortcut
```

### CODE REVIEW

**Review by AshishDhama — COMMENTED** [review]
@hmhealey could you please resolve the conflicts

**Review by jespino — APPROVED** [review]
LGTM

File: components/sidebar/mobile_sidebar_header/contents/contents.tsx [hunk 1]
```diff
@@ -33,31 +30,30 @@ const VerticalStack = styled.div`
...
     teamDisplayName: string;
+    actions: Actions;
+};
+
+type Actions = {
     openModal: <P>(modalData: ModalData<P>) => void;
 };
 
-export default class SidebarHeaderDropdownButton extends React.PureComponent<Props> {
+export default class Contents extends React.PureComponent<Props> {
```
Comment (jespino) [inline components/sidebar/mobile_sidebar_header/contents/contents.tsx:44]: Not for this PR, but this one looks like a good candidate to migrate to functional component. Looks pretty simple and straight forward.

**Review by AshishDhama — APPROVED** [review]

**Comment by hmhealey** [comment]
Looking through those results, the only failure that looked related to this PR passed when I ran it again locally
Screen Shot 2021-12-15 at 2 34 00 PM

**Review by jgilliam17 — COMMENTED** [review]
Thanks @hmhealey 
I added test case IDs and few terminology suggestions 
[E2E report](https://mattermost-cypress-report.s3.amazonaws.com/97688-9740283-onprem-ent-webapp-pr-9512-mm-ee-test:9740283/mochawesome.html) looks good and one test, MM-T1252,  that failed is not failing when testing manually 
Is there anything we can do to make this existing test less flaky?

**Comment by hmhealey** [comment]
No, I'm not sure what we can do to make MM-T1252 more consistent. I'm honestly not sure why it would be failing since it doesn't really check anything until right at the end when it confirms the modal is gone

**Comment by jgilliam17** [comment]
Thanks @hmhealey 
I re-ran the E2Es and beside that one flaky test we discussed, [report](https://mattermost-cypress-report.s3.amazonaws.com/99653-ab10c24-onprem-ent-webapp-pr-9512-mm-ee-test:ab10c24/mochawesome.html) is not showing any PR related failures.
Looks good to merge.

**Review by jgilliam17 — APPROVED** [review]

**Pull request merged**